frontend/src/metabase/dashboard/components/DashCard/DashCard.styled.tsx

```diff
@@ -57,6 +61,12 @@ export const DashboardCardActionsPanel = styled.div<{ isSubMenuOpen: boolean }>`
   // react-resizable covers panel, we have to override it
   z-index: 2;
 
+  // left align on small cards on the left edge to not make the actions go out of the viewport
+  @container (max-width: 200px) {
```

Comment:
200px makes it a good number for cards with 5 buttons, which AFAIK is the max we can have at the moment

https://github.com/metabase/metabase/assets/1914270/b8b3b8a1-dfed-459c-a7ba-f89176c6e750

Reply:
(red color is just to show when the container query applies, same for opacity always at 1 on the actions panel)